eslint.config.mjs

```diff
@@ -0,0 +1,339 @@
+import { fixupConfigRules, fixupPluginRules } from "@eslint/compat";
+import typescriptEslint from "@typescript-eslint/eslint-plugin";
+import prettier from "eslint-plugin-prettier";
+import tsParser from "@typescript-eslint/parser";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import js from "@eslint/js";
+import { FlatCompat } from "@eslint/eslintrc";
+import { createRequire } from "module"
+
+// Workaround for https://github.com/anza-xyz/eslint-plugin-require-extensions/issues/18
+const require = createRequire(import.meta.url)
+const requireExtensions = require("eslint-plugin-require-extensions")
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const compat = new FlatCompat({
+    baseDirectory: __dirname,
+    recommendedConfig: js.configs.recommended,
+    allConfig: js.configs.all
+});
+
+export default [{
+    ignores: [
+        "coverage/**/*",
+        "**/output/**/*",
+        "bundles/**/*",
+        "**/bundle-*",
+        "dist/**/*",
+        "examples/**/*",
+        "test262/**/*",
+        "**/*.html",
+        "**/ava*.config.js",
+        "**/.ava*.config.js",
+        "packages/cosmic-proto/node_modules/",
+        "packages/cosmic-proto/coverage/",
+        "packages/cosmic-proto/dist/",
+        "packages/cosmic-proto/proto/",
+        "packages/cosmic-proto/src/codegen/",
+        "packages/cosmic-proto/scripts/",
+        // Cosmic-swingset specific ignores
+        "packages/cosmic-swingset/dist/**/*",
+        "packages/cosmic-swingset/src/bundles/**/*",
+        "packages/cosmic-swingset/t[0-9]/**/*",
+        "packages/cosmic-swingset/t[0-9].*/**/*",
+        "packages/cosmic-swingset/node_modules/**/*",
+        "packages/cosmic-swingset/build/**/*",
+        // a3p-integration specific ignores
+        "a3p-integration/agoric-sdk/**/*",
+    ],
+}, {
+    // Include both .js and .ts files for casting package
+    files: ["packages/casting/**/*.{js,ts,mjs,cjs}"],
```

Comment:
is there no way to localize this to the casting package directory?

Reply:
With the new flat [config](https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file) style, everything is specified at the root. I didn't see a way to mimic what we were doing with having `.eslintrc` files in specific directories to override behavior